sdks/universal-router-sdk/src/swapRouter.ts

```diff
@@ -2,18 +2,42 @@ import invariant from 'tiny-invariant'
 import { abi } from '@uniswap/universal-router/artifacts/contracts/UniversalRouter.sol/UniversalRouter.json'
 import { Interface } from '@ethersproject/abi'
 import { BigNumber, BigNumberish } from 'ethers'
-import { MethodParameters } from '@uniswap/v3-sdk'
+import {
+  MethodParameters,
+  Position as V3Position,
+  NonfungiblePositionManager as V3PositionManager,
+  RemoveLiquidityOptions as V3RemoveLiquidityOptions,
+} from '@uniswap/v3-sdk'
+import {
+  Position as V4Position,
+  V4PositionManager,
+  AddLiquidityOptions as V4AddLiquidityOptions,
+  MintOptions,
+} from '@uniswap/v4-sdk'
 import { Trade as RouterTrade } from '@uniswap/router-sdk'
-import { Currency, TradeType } from '@uniswap/sdk-core'
+import { Currency, TradeType, Percent, CHAIN_TO_ADDRESSES_MAP, SupportedChainsType } from '@uniswap/sdk-core'
 import { UniswapTrade, SwapOptions } from './entities/actions/uniswap'
-import { RoutePlanner } from './utils/routerCommands'
-import { encodePermit } from './utils/inputTokens'
+import { RoutePlanner, CommandType } from './utils/routerCommands'
+import { encodePermit, encodeV3PositionPermit, V3PositionPermit } from './utils/inputTokens'
+import { UNIVERSAL_ROUTER_ADDRESS, UniversalRouterVersion } from './utils/constants'
 
 export type SwapRouterConfig = {
   sender?: string // address
   deadline?: BigNumberish
 }
 
+export interface MigrateV3ToV4Options {
+  inputPosition: V3Position
+  outputPosition: V4Position
+  v3RemoveLiquidityOptions: V3RemoveLiquidityOptions
+  v4AddLiquidityOptions: V4AddLiquidityOptions
+  inputV3NFTPermit?: V3PositionPermit
```

Comment:
ah sorry I didn't catch this earlier but I think that this is an unneeded parameter. the v3RemoveLiquidityOptions defined already an optional `permit` field